lib/compress.js

```diff
@@ -176,6 +176,21 @@ merge(Compressor.prototype, {
         }
     };
 
+    // we shouldn't compress (1,func)(something) to
+    // func(something) because that changes the meaning of
+    // the func (becomes lexical instead of global).
+    function check_function_call(parent, orig, val) {
```

Comment:
@michaelficarra It's the end effect that is being described. I'd also be okay with `fix_call_expression`.

Side note, you seem to be familiar with the ECMAscript specs... do you happen to know the section(s) in the ES 5.1 spec that outlines this behavior?